### DISCUSSION TITLE
Releationship is ignoring where in the relations

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#630] [Q&A (Help)] [answered]
`InstituteClass`

    /**
     * @return BelongsToMany
     */
    public function teachers(): BelongsToMany
    {
        return $this->belongsToMany(User::class)
            ->wherePivot('role', User::ROLE_TEACHER)
        ;
    }

`InstituteClassController`

        CRUD::field('teachers')
            ->type('relationship')
            ->label('Isegnanti')
            ->entity('teachers')
            ->pivotSelect([
                'name' => 'Insegnante',
            ])
            ->subfields([
                [
                   'name' => 'role',
                   'type' => 'hidden',
                   'value' => User::ROLE_TEACHER,
                ]
            ]);
        ;

The problem is I see in the select ALL the user NOT ONLY the one with 'techer' as pivot role table.

How can I prefilter these entries?
Ho can it ignores the wherePivot in the relation?

**Comment by pxpm** [comment, marked as answer]
@realtebo we use the model to get the options, you add the constrain in the relation for the pivot table, not the model.

You can configure the options like in any other select: 
```php
'pivotSelect' => [
    'options' => function($model) {
        return $model->whereIn('id', [1,2]);
    },
],
```
Cheers

**Reply by realtebo** [reply]
Please DOCUMENT IT

**Reply by pxpm** [reply]
Hello @realtebo it's not good for anyone that you use that tone. 
I am here helping you the best I can. 
I can document it better for sure, make it easy for everyone, but I don't think yelling will make it better for any of us involved here. 

I think we have a pretty open mind with our clients/users. I am here helping you that brought pro, as I am here helping people that didn't spent a cent in backpack. 
We do it from the heart because we believe that our users are the best feedback, and we have always been open to accept your suggestions and help you when you asked for. 

I am sure your intentions are good, but please don't use this tone. 

Sorry if I am getting it wrong. From my understanding we have it documented in the select field and we say that the pivot select can be configured like any other select.

Let's try to be mindful of that in future. 

: 🙏

**Reply by realtebo** [reply]
I apologize. Sorry. I forgot that uppercase letters are like screaming.

I started with a "please", because I was asking something that is not mandatory,
So, sorry, again, and thanks for all your work.


I appreciated your reminder .

**Reply by pxpm** [reply]
Sure @realtebo I understand the frustration when we face some bug, or find something not properly documented. You help me reporting it, I help you answering and improving our software. Win win. Appreciate your efforts reporting stuff and asking questions, we have improved alot thanks your feedback: 🙏
Let me know if I can help you with something else . 

Cheers

**Comment by pxpm** [comment]
I've added more examples for `pivotSelect` and comments that I hope will be more helpful for everyone. https://github.com/Laravel-Backpack/docs/pull/495

Also added a title for it to make it easy to find while searching in the docs.

Thanks 🙏
